### PR TITLE
airbyte-ci: tweaks for cross-repo compatibility

### PR DESCRIPTION
**Pull request by postamar**
## What
This PR is a series of harmless changes which make it possible to re-use most of the tooling in this git repository in another git repository.

## Review guide
Go through the PR one commit at a time.

## User Impact
None whatsoever.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/build.gradle [hunk 1]
```diff
@@ -60,9 +60,6 @@ tasks.named('clean').configure {
 }
 
 allprojects {
-    // Evaluate CDK project before evaluating the connector.
-    evaluationDependsOn(':airbyte-cdk:java:airbyte-cdk')
-
```
Comment (postamar) [inline airbyte-integrations/connectors/build.gradle:65]: this isn't actually required at all in any case

File: airbyte-ci/connectors/connector_ops/connector_ops/utils.py [hunk 2]
```diff
@@ -207,26 +207,6 @@ def parse_gradle_dependencies(build_file: Path) -> Tuple[List[Path], List[Path]]
...
-    """
-    base_path = Path("airbyte-cdk/java/airbyte-cdk")
-    found: List[Path] = [base_path]
-    for submodule in ["core", "db-sources", "db-destinations"]:
-        found.append(base_path / submodule)
-        project_dependencies, test_dependencies = parse_gradle_dependencies(base_path / Path(submodule) / Path("build.gradle"))
-        found += project_dependencies
-        if with_test_dependencies:
-            found += test_dependencies
-    return list(set(found))
```
Comment (postamar) [inline airbyte-ci/connectors/connector_ops/connector_ops/utils.py:227]: this function has been broken for a while

File: airbyte-ci/connectors/connector_ops/connector_ops/utils.py [hunk 3]
```diff
@@ -242,12 +222,6 @@ def get_all_gradle_dependencies(
     if found_dependencies is None:
         found_dependencies = []
     project_dependencies, test_dependencies = parse_gradle_dependencies(build_file)
-
-    # Since first party project folders are transitive (compileOnly) in the
-    # CDK, we always need to add them as the project dependencies.
-    project_dependencies += get_local_cdk_gradle_dependencies(False)
-    test_dependencies += get_local_cdk_gradle_dependencies(with_test_dependencies=True)
```
Comment (postamar) [inline airbyte-ci/connectors/connector_ops/connector_ops/utils.py:249]: CDK projects haven't been compileOnly for a while

File: airbyte-ci/connectors/connector_ops/connector_ops/utils.py [hunk 4]
```diff
@@ -648,6 +622,7 @@ def __repr__(self) -> str:
     def get_local_dependency_paths(self, with_test_dependencies: bool = True) -> Set[Path]:
         dependencies_paths = []
         if self.language == ConnectorLanguage.JAVA:
+            dependencies_paths += [Path("./airbyte-cdk/java/airbyte-cdk")]
```
Comment (postamar) [inline airbyte-ci/connectors/connector_ops/connector_ops/utils.py:625]: This imports all of the CDK source, which isn't that big in the first place, so this should be OK. This is required when connectors are built with `useLocalCdk.= true`.

**Comment by alafanechere** [comment]
Let's run a test workflow on a Java an python connectors to make sure it does not break anything:
https://github.com/airbytehq/airbyte/actions/runs/9890550942

**Comment by postamar** [comment]
Good call. I forgot to mention that I tested this on my machine. Testing on a github runner is better. Thanks for taking the initiative!

**Review by alafanechere — APPROVED** [review]

**Pull request merged**